src/NotEmpty.php

```diff
@@ -100,19 +100,8 @@ public function __construct($options = null)
         }
 
         if (is_array($options)) {
-            if (!array_key_exists('type', $options)) {
-                $detected = 0;
-                $found    = false;
-                foreach ($options as $option) {
-                    if (in_array($option, $this->constants, true)) {
-                        $found = true;
-                        $detected += array_search($option, $this->constants);
-                    }
-                }
-
-                if ($found) {
-                    $options['type'] = $detected;
-                }
+            if (($type = $this->calculateTypeValue($options)) != 0) {
+                $options['type'] = $type;
```

Comment:
This would break options for AbstractValidator. I checked and it doesn't actually cause testTypeAutoDetectionHasNoSideEffect to fail, so maybe another test should be added that uses an array that has both types and AbstractValidator options? Either that or document that if the array has any non-type values it will just be treated as a normal options array.
It looks like the current constructor allows type strings to be combined with other options.